tests/testproject/TestProject.AppHost/TestProgram.cs

```diff
@@ -160,51 +155,5 @@ public void Run()
     }
 
     public void Dispose() => App?.Dispose();
-
-    /// <summary>
-    /// Writes the allocated endpoints to the console in JSON format.
-    /// This allows for easier consumption by the external test process.
-    /// </summary>
-    private sealed class EndPointWriterHook : IDistributedApplicationEventingSubscriber
-    {
-        public async Task OnAfterResourcesCreated(AfterResourcesCreatedEvent @event, CancellationToken cancellationToken)
-        {
-            var root = new JsonObject();
-            foreach (var project in @event.Model.Resources.OfType<ProjectResource>())
-            {
-                var projectJson = new JsonObject();
-                root[project.Name] = projectJson;
-
-                var endpointsJsonArray = new JsonArray();
-                projectJson["Endpoints"] = endpointsJsonArray;
-
-                foreach (var endpoint in project.Annotations.OfType<EndpointAnnotation>())
-                {
-                    var allocatedEndpoint = endpoint.AllocatedEndpoint;
-                    if (allocatedEndpoint is null)
-                    {
-                        continue;
-                    }
-
-                    var endpointJsonObject = new JsonObject
-                    {
-                        ["Name"] = endpoint.Name,
-                        ["Uri"] = allocatedEndpoint.UriString
-                    };
-                    endpointsJsonArray.Add(endpointJsonObject);
-                }
-            }
-
-            // write the whole json in a single line so it's easier to parse by the external process
-            await Console.Out.WriteLineAsync("$ENDPOINTS: " + JsonSerializer.Serialize(root, JsonSerializerOptions.Default));
-        }
-
-        public Task SubscribeAsync(IDistributedApplicationEventing eventing, DistributedApplicationExecutionContext executionContext, CancellationToken cancellationToken)
-        {
-            // We can assume endpoints are allocated before project resources are created
-            eventing.Subscribe<AfterResourcesCreatedEvent>(OnAfterResourcesCreated);
-            return Task.CompletedTask;
-        }
-    }
```

Comment:
This cannot be dropped. It is used by template tests and EndToEnd tests via https://github.com/dotnet/aspire/blob/main/tests/Shared/TemplatesTesting/AspireProject.cs